File: src/types/verification.ts
import type { DocumentStatus } from './document'

export interface VerificationIssuer {
  id: string
  name: string
}

export type VerificationOutcome = 'VALID' | 'REVOKED' | 'EXPIRED' | 'NOT_FOUND' | 'INVALID'

export interface VerificationDocumentSummary {
  id: string
  number: string
  title: string
  type: string
  holderName: string
  issuedAt: string
  issuer: VerificationIssuer
}

export interface VerificationSignature {
  valid: boolean
  algorithm: string
  signedBy: string
  signedAt: string
}

export interface VerificationDetails {
  valid: boolean
  outcome: VerificationOutcome
  status: DocumentStatus
  reference: string
  verifiedAt: string
  document: VerificationDocumentSummary
  signature?: VerificationSignature
  message: string
}

export interface VerificationQuery {
  code?: string
  documentId?: string
}

File: backend/app/schemas/verification.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict


class VerificationSignatureInfo(BaseModel):
    valid: bool
    algorithm: str | None = None


class VerificationInstitutionInfo(BaseModel):
    id: str
    legal_name: str
    country: str | None = None
    website: str | None = None


class VerificationDocumentInfo(BaseModel):
    number: str
    type: str
    title: str
    holder_name: str
    description: str | None = None
    issued_at: datetime
    expires_at: datetime | None = None
    status: str


class VerificationResult(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    code: str
    status: str
    message: str
    document: Optional["VerificationDocumentInfo"] = None
    institution: VerificationInstitutionInfo | None = None
    signature: VerificationSignatureInfo
    verified_at: datetime


VerificationResult.model_rebuild()
